src/Compilers/CSharp/Test/Semantic/Semantics/ExtensionTypeTests.cs

```diff
@@ -8158,4 +8232,1151 @@ public void ObsoleteOnType()
+        comp.VerifyDiagnostics(
+            // (2,39): error CS0426: The type name 'Garbage' does not exist in the type 'B'
+            // explicit extension B for object : A<B.Garbage> { }
+            Diagnostic(ErrorCode.ERR_DottedTypeNameNotFoundInAgg, "Garbage").WithArguments("Garbage", "B").WithLocation(2, 39)
+            );
+    }
+
+    [Fact]
+    public void UseSiteErrorReporting_MissingGrandBase()
+    {
+        var source1 = """
+public explicit extension A for object { }
+""";
+        var compilation1 = CreateCompilation(source1, assemblyName: "missing",
+            targetFramework: TargetFramework.Net70);
+
+        compilation1.VerifyDiagnostics();
+
+        var source2 = """
+public explicit extension B for object : A { }
+public explicit extension C<T> for object : B { }
+""";
+        var compilation2 = CreateCompilation(source2, references: new[] { compilation1.ToMetadataReference() },
+            targetFramework: TargetFramework.Net70);
+
+        compilation2.VerifyDiagnostics();
+
+        var source3 = """
+D.M();
+
+public explicit extension D for object : C<string> 
+{
+    public static void M() { }
+}
+""";
+        var compilation3 = CreateCompilation(source3, references: new[] { compilation2.ToMetadataReference() },
+            targetFramework: TargetFramework.Net70);
+
+        compilation3.VerifyDiagnostics(
+            // (1,1): error CS8090: There is an error in a referenced assembly 'missing, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
+            // D.M();
+            Diagnostic(ErrorCode.ERR_ErrorInReferencedAssembly, "D.M").WithArguments("missing, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null").WithLocation(1, 1),
+            // (1,3): error CS8090: There is an error in a referenced assembly 'missing, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
+            // D.M();
+            Diagnostic(ErrorCode.ERR_ErrorInReferencedAssembly, "M").WithArguments("missing, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null").WithLocation(1, 3),
+            // (3,27): error CS8090: There is an error in a referenced assembly 'missing, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
+            // public explicit extension D for object : C<string> 
+            Diagnostic(ErrorCode.ERR_ErrorInReferencedAssembly, "D").WithArguments("missing, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null").WithLocation(3, 27)
+            );
+    }
+
+    [Fact]
+    public void MemberLookup_MethodsFromObject()
+    {
+        var src = """
+void M(A a)
+{
+    A.ToString();
+    a.ToString();
+}
+
+public explicit extension A for object { }
+""";
+        // PROTOTYPE methods from System.Object will come from member access (not from member lookup)
+        var comp = CreateCompilation(src, targetFramework: TargetFramework.Net70);
+        comp.VerifyDiagnostics(
+            // (1,6): warning CS8321: The local function 'M' is declared but never used
```

Comment:
>// (1,6): warning CS8321: The local function 'M' is declared but never used [](http://example.com/codeflow?start=12&length=75)

Could we suppress this warning (it is just a noise), or use a local instead of the parameter?